routing/router.go

```diff
@@ -616,373 +450,156 @@ func (r *ChannelRouter) networkHandler() {
-		// to be fully verified once we advance forward in the chain.
-		if isPremature(&msg.ChannelID) {
-			blockHeight := msg.ChannelID.BlockHeight
-			log.Infof("Update announcement for chan_id=(%v), is "+
-				"premature: advertises height %v, only height "+
-				"%v is known", chanID, blockHeight,
-				r.bestHeight)
-
-			r.prematureAnnouncements[blockHeight] = append(
-				r.prematureAnnouncements[blockHeight],
-				msg,
-			)
-			return false
 		}
 
-		// As edges are directional edge node has a unique policy for
-		// the direction of the edge they control. Therefore we first
+		// As edges are directional msg msg has a unique policy for
+		// the direction of the msg they control. Therefore we first
 		// check if we already have the most up to date information for
-		// that edge. If so, then we can exit early.
-		updateTimestamp := time.Unix(int64(msg.Timestamp), 0)
+		// that msg. If so, then we can exit early.
 		switch msg.Flags {
 
 		// A flag set of 0 indicates this is an announcement for the
-		// "first" node in the channel.
+		// "first" msg in the channel.
 		case 0:
-			if edge1Timestamp.After(updateTimestamp) ||
-				edge1Timestamp.Equal(updateTimestamp) {
+			if edge1Timestamp.After(msg.LastUpdate) ||
+				edge1Timestamp.Equal(msg.LastUpdate) {
+
+				return errors.Errorf("ignoring announcement "+
+					"(flags=%v) for known chan_id=%v", msg.Flags,
+					msg.ChannelID)
 
-				log.Debugf("Ignoring announcement (flags=%v) "+
-					"for known chan_id=%v", msg.Flags,
-					chanID)
-				return false
 			}
 
 		// Similarly, a flag set of 1 indicates this is an announcement
-		// for the "second" node in the channel.
+		// for the "second" msg in the channel.
 		case 1:
-			if edge2Timestamp.After(updateTimestamp) ||
-				edge2Timestamp.Equal(updateTimestamp) {
-
-				log.Debugf("Ignoring announcement (flags=%v) "+
-					"for known chan_id=%v", msg.Flags,
-					chanID)
-				return false
+			if edge2Timestamp.After(msg.LastUpdate) ||
+				edge2Timestamp.Equal(msg.LastUpdate) {
+				return errors.Errorf("ignoring announcement "+
+					"(flags=%v) for known chan_id=%v", msg.Flags,
+					msg.ChannelID)
 			}
 		}
 
-		// Before we can update the channel information, we need to get
+		// Before we can msg the channel information, we need to get
 		// the UTXO itself so we can store the proper capacity.
-		chanPoint, err := r.fetchChanPoint(&msg.ChannelID)
+		chanPoint, err := r.fetchChanPoint(&channelID)
 		if err != nil {
-			log.Errorf("unable to fetch chan point for chan_id=%v: %v", chanID, err)
-			return false
+			return errors.Errorf("unable to fetch chan point for "+
+				"chan_id=%v: %v", msg.ChannelID, err)
 		}
 		if _, err := r.cfg.Chain.GetUtxo(&chanPoint.Hash,
 			chanPoint.Index); err != nil {
-			log.Errorf("unable to fetch utxo for chan_id=%v: %v",
-				chanID, err)
-			return false
+			return errors.Errorf("unable to fetch utxo for "+
+				"chan_id=%v: %v", msg.ChannelID, err)
```

Comment:
With the `discovery` package handling full validation, this fragment can be removed all together as the `ChannelRouter` should only receive fully valid data.